scripts/common/config-item.ts

```diff
@@ -1,26 +1,98 @@
+import { PartialRecord } from '@helpers/types/common'
 import { ContractNames } from '@oasisdex/oasis-actions/src'
 import { Address } from '@oasisdex/oasis-actions/src/types'
 
 export type ConfigItem = {
-  name: string
+  name: AllowedContractNames
   serviceRegistryName?: ContractNames
   address: Address
 }
 
 export type SystemConfigItem = ConfigItem & {
+  name: DeployedSystemContractNames
   deploy: boolean
   history: Address[]
   constructorArgs?: Array<number | string>
 }
 
+type SwapContractNames = 'Swap' | 'uSwap'
+export type CoreContractNamesWithoutSwap =
+  | 'ServiceRegistry'
+  | 'OperationExecutor'
+  | 'OperationStorage'
+  | 'OperationsRegistry'
+  | 'DSProxyFactory'
+  | 'DSProxyRegistry'
+  | 'DSGuardFactory'
+  | 'AccountGuard'
+  | 'AccountFactory'
+  | 'ChainLogView'
+export type CoreContractNames = CoreContractNamesWithoutSwap | SwapContractNames
+export type AaveV2ContractNames = 'AaveBorrow' | 'AaveDeposit' | 'AaveWithdraw' | 'AavePayback'
+export type ActionContractNamesWithoutAaveV2 =
+  | 'SwapAction'
+  | 'PullToken'
+  | 'SendToken'
+  | 'SetApproval'
+  | 'WrapEth'
+  | 'UnwrapEth'
+  | 'TakeFlashloan'
+  | 'ReturnFunds'
+  | 'PositionCreated'
+  | 'AaveV3Borrow'
+  | 'AaveV3Deposit'
+  | 'AaveV3Withdraw'
+  | 'AaveV3Payback'
+  | 'AaveV3SetEMode'
+export type ActionContractNames = ActionContractNamesWithoutAaveV2 | AaveV2ContractNames
+type CommonContractNames =
+  | 'WETH'
+  | 'ETH'
+  | 'STETH'
+  | 'WSTETH'
+  | 'DAI'
+  | 'USDC'
+  | 'WBTC'
+  | 'UniswapRouterV3'
+  | 'FlashMintModule'
+  | 'BalancerVault'
+  | 'OneInchAggregator'
+  | 'AuthorizedCaller'
+  | 'FeeRecipient'
+  | 'ChainlinkEthUsdPriceFeed'
+type AaveProtocolContractNamesV2 =
+  | 'PriceOracle'
+  | 'LendingPool'
+  | 'ProtocolDataProvider'
+  | 'WETHGateway'
+type AaveProtocolContractNamesV3 = 'AaveOracle' | 'Pool' | 'AaveProtocolDataProvider'
+type AaveProtocolContractNames = AaveProtocolContractNamesV2 | AaveProtocolContractNamesV3
+
+export type AllowedContractNames =
+  | CoreContractNames
+  | ActionContractNames
+  | CommonContractNames
+  | AaveProtocolContractNames
+
+export type LocalSystemContractNames = 'DsProxyRegistry'
```

Comment:
Why do we need this per se?

Reply:
We don't. Relic from before merging your stuff